File: test6.java
class ErrorHandler {
    public static void main(String[] args) {
        try {
            errorHappens();
        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        }
    }

    private static void errorHappens() throws Exception {
        throw new Exception("This is an error");
    }
}

File: test6.php
<?php

namespace Hesper;

/**
 * Write a method in the errorHandler class that calls the errorsHappen method and catches its error
 * and echos the resulting message with an "Error: " string before it.
 *
 * Expected output: "Error: This is an error"
 *
 * Make sure to execute your class method so that when we run php test6.php we get the expected output
 *
 */

class errorHandler {
    private function errorsHappen() {
        throw new \Exception('This is an error');
    }

    function imminentErrorMethod() {
        try {
            errorsHappen();
        } catch (Exception $e) {
            echo 'Error: ', $e->getMessage(), "\n";
        }
    }
}

$errorHandler = new errorHandler();
$errorHandler->imminentErrorMethod();

